cmds/truncate/truncate.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Truncate - shrink or extend the size of a file to the specified size
+//
+// Synopsis:
+//     truncate [OPTIONS] [FILE]...
+//
+// Options:
+//     -s: size in bytes
+//     -c: do not create any files
+//
+// Author:
+//     Roland Kammerer <dev.rck@gmail.com>
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+)
+
+var (
+	size     = flag.Int64("s", -1, "size in bytes")
+	nocreate = flag.Bool("c", false, "do not create files")
+)
+
+func main() {
+	flag.Parse()
+
+	if *size == -1 {
+		log.Fatalln("truncate: ERROR: You have to specify a non-negative size")
+	}
+
+	if flag.NArg() == 0 {
+		log.Fatalln("truncate: ERROR: missing file name.")
+	}
+
+	for _, fname := range flag.Args() {
+		if _, err := os.Stat(fname); os.IsNotExist(err) && !*nocreate {
+			if f, err := os.Create(fname); err != nil {
+				log.Fatalf("truncate: ERROR: %v\n", err)
+			} else {
+				f.Close()
+			}
+		}
+
+		if err := os.Truncate(fname, *size); err != nil {
```

Comment:
The standard GNU util doesn't fail on errors. Try this:
rm -f /tmp/y
truncate -c -s 2 /tmp/y
no error. To match the GNU tool we have to at least ignore errors or at most print a message.
truncate is kind of a lousy tool. It creates, but does not let you specify the mode.